### PR TITLE
Onboarding softwareplan for autogeneration

### PR DESCRIPTION
**Pull request by hunmix**
```
> azure-schema-generator@1.0.0 generate-single D:\code\azure-resource-manager-schemas\generator
> ts-node cmd/generatesingle "softwareplan/resource-manager"

[C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc] executing: git fsck
[C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc] executing: git reset -q .
[C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc] executing: git checkout -q -- .
[C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc] executing: git clean -q -fd
[C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc] executing: git gc
[C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc] executing: git fetch -q
[C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc] executing: git checkout -q origin/master
Using whitelist config:
{
  "basePath": "softwareplan/resource-manager",
  "namespace": "Microsoft.SoftwarePlan",
  "resourceConfig": [
    {
      "type": "hybridUseBenefits",
      "scopes": 32
    }
  ]
}
[D:\code\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --version=3.0.6242 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=C:\Users\ADMINI~1\AppData\Local\Temp\ylrdtsikmfe --tag=all-api-versions --api-version=2019-06-01-preview --title=none --pass-thru:subset-reducer C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc\specification\softwareplan\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v12.16.1, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\Administrator\.autorest\@autorest_core@3.0.6242\node_modules\@autorest\core\dist' (3.0.6242)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[4.41 s] Generation Complete
================================================================================================================================
Filename: D:\code\azure-resource-manager-schemas\schemas\2019-06-01-preview\Microsoft.SoftwarePlan.json
Provider Namespace: Microsoft.SoftwarePlan
API Version: 2019-06-01-preview
Resource Types (Extension Scope):
- hybridUseBenefits
================================================================================================================================
[D:\code\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --version=3.0.6242 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=C:\Users\ADMINI~1\AppData\Local\Temp\pm93xk6nztf --tag=all-api-versions --api-version=2019-12-01 --title=none --pass-thru:subset-reducer C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc\specification\softwareplan\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v12.16.1, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\Administrator\.autorest\@autorest_core@3.0.6242\node_modules\@autorest\core\dist' (3.0.6242)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[4.28 s] Generation Complete
================================================================================================================================
Filename: D:\code\azure-resource-manager-schemas\schemas\2019-12-01\Microsoft.SoftwarePlan.json
Provider Namespace: Microsoft.SoftwarePlan
API Version: 2019-12-01
Resource Types (Extension Scope):
- hybridUseBenefits
================================================================================================================================
```

### CODE REVIEW

File: generator/whitelist.ts [hunk 1]
```diff
@@ -314,6 +314,16 @@ const whitelist: WhitelistConfig[] = [
         basePath: 'storagesync/resource-manager',
         namespace: 'Microsoft.StorageSync',
     },
+    {
+        basePath: 'softwareplan/resource-manager',
+        namespace: 'Microsoft.StorSimple',
+        resourceConfig: [
+            {
+                type: 'hybridUseBenefits',
+                scopes: ScopeType.ResourceGroup,
```
Comment (anthony-c-martin) [inline generator/whitelist.ts]: This should be `ScopeType.Extension`. Please could you change this and re-run generation?

**Comment by hunmix** [comment]
```
> azure-schema-generator@1.0.0 generate-single D:\code\azure-resource-manager-schemas\generator
> ts-node cmd/generatesingle "softwareplan/resource-manager"

[C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc] executing: git fsck
[C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc] executing: git reset -q .
[C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc] executing: git checkout -q -- .
[C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc] executing: git clean -q -fd
[C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc] executing: git gc
[C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc] executing: git fetch -q
[C:\Users\ADMINI~1\AppData\Local\Temp\schm_azspc] executing: git checkout -q origin/master
Using whitelist config:
{
  "basePath": "softwareplan/resource-manager",
  "namespace": "Microsoft.StorSimple",
  "resourceConfig": [
    {
      "type": "hybridUseBenefits",
      "scopes": 32
    }
  ]
}
```

File: generator/whitelist.ts [hunk 2]
```diff
@@ -326,10 +326,21 @@ const whitelist: WhitelistConfig[] = [
         basePath: 'sql/resource-manager',
         namespace: 'Microsoft.Sql',
     },
+    {
+        basePath: 'softwareplan/resource-manager',
+        namespace: 'Microsoft.StorSimple',
```
Comment (anthony-c-martin) [inline generator/whitelist.ts]: This should be `Microsoft.SoftwarePlan`

File: generator/whitelist.ts [hunk 3]
```diff
@@ -326,10 +326,21 @@ const whitelist: WhitelistConfig[] = [
...
+                type: 'hybridUseBenefits',
+                scopes: ScopeType.Extension,
+            },
+        ]
+    },
     {
         basePath: 'timeseriesinsights/resource-manager',
         namespace: 'Microsoft.TimeSeriesInsights',
     },
+
```
Comment (anthony-c-martin) [inline generator/whitelist.ts]: [nit] remove the new line here

**Pull request merged**